apps/dapp-console-api/src/utils/Faucet.ts

```diff
@@ -0,0 +1,195 @@
+import type { Address, Hex, TypedDataDomain } from 'viem'
+import { getContract, keccak256, numberToHex } from 'viem'
+import { sepolia } from 'viem/chains'
+
+import { faucetAbi } from '../abi/Faucet'
+import {
+  sepoliaAdminWalletClient,
+  sepoliaPublicClient,
+} from '../constants/faucetConfigs'
+import { getFaucetContractBalance } from './faucetBalances'
+import type { RedisCache } from './redis'
+
+export type FaucetConstructorArgs = {
+  chainId: number
+  redisCache: RedisCache
+  faucetAddress: Address
+  onChainAuthModuleAddress: Address
+  offChainAuthModuleAddress: Address
+  displayName: string
+  onChainDripAmount: bigint
+  offChainDripAmount: bigint
+  blockExplorerUrl?: string
+}
+
+export const faucetAuthModes = [
+  'GITHUB',
+  'OPTIMIST_NFT',
```

Comment:
we wont hae optimist nft as an auth method anymore, right? if so, we should remove this